packages/zoe/src/zoeService/makeInvitation.js

```diff
@@ -20,31 +29,41 @@ export const createInvitationKit = shutdownZoeVat => {
     assert.typeof(installation, 'object');
 
     /** @type {ZoeInstanceAdminMakeInvitation} */
-    const makeInvitation = (
+    const makeInvitation = async (
       invitationHandle,
       description,
       customProperties,
+      relativeFee = undefined,
+      relativeExpiry = undefined,
     ) => {
       assert.typeof(invitationHandle, 'object');
       assert.typeof(
         description,
         'string',
         X`The description ${description} must be a string`,
       );
+      const absoluteFee = translateFee(relativeFee);
+      const absoluteExpiry = await translateExpiry(relativeExpiry);
+
+      let feeInfo = {};
+      if (absoluteFee && absoluteExpiry) {
+        feeInfo = { fee: absoluteFee, expiry: absoluteExpiry };
+      }
       // If the contract-provided customProperties include the
-      // properties 'description', 'handle', 'instance' and
-      // 'installation', their corresponding values will be
-      // overwritten with the actual values. For example, the value
-      // for `instance` will always be the actual instance for the
-      // contract, even if customProperties includes a property called
-      // `instance`.
+      // properties 'description', 'handle', 'instance',
+      // 'installation', 'fee', or 'expiry', their corresponding
+      // values will be overwritten with the actual values. For
+      // example, the value for `instance` will always be the actual
+      // instance for the contract, even if customProperties includes
+      // a property called `instance`.
       const invitationAmount = AmountMath.make(invitationKit.brand, [
         {
           ...customProperties,
           description,
           handle: invitationHandle,
           instance,
           installation,
+          ...feeInfo, // may be empty
```

Comment:
I followed the translateExpiry and see it returns a `Timestamp` which is a raw bigint. Which is meaningless to Fred without knowing what `timeAuthority` it is relative to. That's why Fred needs it in the coveredCall.

Since you already define an explicit `Timestamp` type, that currently aliases to `bigint`, would redefining it be a good place to take @dtribble 's suggestion of passing around a number-timeAuthority pair, like we do amounts? Whether we do it in the type of not, consider passing a pair record here rather than just adding `timeAuthority` to the authoritative property names.

Reply:
Ah, good point. My idea here was that Fred could call E(zoe).getConfiguration() to get the Zoe timeAuthority, so there would be no confusion about what timeAuthority was being used. I think that takes care of any Fred problem. When we rethink how metering works in general, we should revisit this to see if it can be easily improved.

Reply:
I'll add a new property `zoeTimeAuthority` to the invitation details to solve this, like we discussed.